internal/provider/provider.go

```diff
@@ -377,6 +380,23 @@ func (p *fluxProvider) Configure(ctx context.Context, req provider.ConfigureRequ
 		resp.Diagnostics.AddError("Could not create provider resource data", err.Error())
 		return
 	}
+
+	// ================================================
+	// Validate kubernetes connectivity
+	// ================================================
+	kubeClient, err := prd.GetKubernetesClient()
+	if err != nil {
+		resp.Diagnostics.AddError("Could not create Kubernetes client", err.Error())
+		return
+	}
+
+	var list corev1.ServiceAccountList
```

Comment:
```suggestion
	var list apiextensionsv1.CustomResourceDefinitionList
```

Using this import `apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"`

This is what we do in the Flux CLI, which also verifies that the kubeconfig user has cluster-wide access which is required to install Flux.

Reply:
@stefanprodan Is there a better way for me to use flux to verify connectivity?

Reply:
The code I suggest should be enough for validating both the  connectivity and permissions.